### PR TITLE
fix RESTlet types, add CustomRecordActionScript types

### PR DESCRIPTION
**Pull request by GabrielGorta**

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -5,13 +5,12 @@
...
     "sourceMap": false,
     "newLine": "LF",
+    "esModuleInterop": true,
+    "allowSyntheticDefaultImports": true,
     "experimentalDecorators": true,
     "noImplicitUseStrict": true,
     "baseUrl": ".",
-    "lib": ["es5", "es2015.promise", "dom"],
-    "paths": {
-      "N/*": ["N/*"]
```
Comment (GabrielGorta) [inline tsconfig.json:13]: this casues problems with imports, not sure why this is there, but it doesn't make sense to me.

File: package.json [hunk 2]
```diff
@@ -9,7 +9,6 @@
   },
   "homepage": "https://github.com/headintheclouddev/typings-suitescript-2.0",
   "version": "2022.2.3",
-  "main": "index.d.ts",
```
Comment (GabrielGorta) [inline package.json:12]: doesn't exist

**Comment by MrRob** [comment]
What is Custom Record Action script?  Is that something new in 2022.2?

**Comment by GabrielGorta** [comment]
it's since _2020.2_

it's in options in SuiteCloud extension: https://docs.oracle.com/en/cloud/saas/netsuite/ns-online-help/section_162810635242.html

What it is for? I don't know either. Never used it. But it exist, so I think it should be in types too.

**Pull request merged**